app/controllers/api/v1/materials_controller.rb

```diff
@@ -29,11 +29,33 @@ def create
 
   # PATCH/PUT /materials/1
   def update
-    if @material.update(material_params)
-      render json: @material
-    else
+    material_type = @material.material_type
+    unless material_type_update_params[:relationships].nil? or material_type_update_params[:relationships][:material_type].nil?
+      material_type = MaterialType.find_by(material_type_update_params[:relationships][:material_type][:data][:attributes])
+    end
+
+    begin
+      ActiveRecord::Base.transaction do
+        if metadata_update_params[:relationships] and metadata_update_params[:relationships][:metadata]
+          metadata_update_params[:relationships][:metadata][:data].each do |new_metadatum|
+            metadatum = @material.metadata.find{ |metadatum| metadatum.key == new_metadatum[:attributes][:key] }
+            if metadatum.nil?
+              @material.metadata << Metadatum.new(new_metadatum[:attributes])
+            else
+              metadatum.value = new_metadatum[:attributes][:value]
```

Comment:
I would suggest using update_attributes() to update :value,  to be sure we update the value for metadatum in the database record.